### ISSUE TITLE
[bug] ANY keyword does not work for os.distro in settings.yml

### ISSUE THREAD

**Issue by Simonlovald** [conan-io/conan#15414]
### Environment details

* Operating System+version: Linux, RHEL 8.6
* Compiler+version: gcc 9.3
* Conan version: 2.0.16
* Python version: 3.10.

### Steps to reproduce

In settings.yml add ANY as a key together with other specific distro like this:
```    
os:
     Linux:
          distro:
              ANY:
                  version: [ANY]
 ```
 When trying to use a profile with a specific distro, e.g. "CentOS", then it returns KeyError: 'CentOS'

**Comment by memsharded** [comment]
Hi @Simonlovald 

Thanks for your report.
I am afraid this is not a bug so far, but by design. ``ANY`` is only designed to serve as a final possible value with the ``[ANY]`` syntax, but not to define new categories with nested subsettings. I am having a look to check if this could make sense.

**Comment by memsharded** [comment]
One of the reasons this syntax hasn't been allowed so far is because it introduces some issues in the definition and expectations of ANY if we define for example:

```yaml
os:
     Linux:
          distro:
              ANY:
                  version: [ANY]
              CentOS:
                  version: [1, 2, 3]
              Ubuntu:
                  version: ["16.04", ...]
```
Which seems kind of natural definition and expected behavior, but it can be challenging to implement and manage.

**Comment by memsharded** [comment]
It is also true that ``Settings`` got some internal refactors and improvements in Conan 2.0, and it seems more doable now. I am exploring the idea (no guarantees yet) in https://github.com/conan-io/conan/pull/15415, to be discussed with the team

**Comment by Simonlovald** [comment]
Great, thank you @memsharded. I would suggest that in the meantime the [description in the docs](https://docs.conan.io/2.0/reference/config_files/settings.html) for the settings.yml is updated. Right now it states: "If a setting is allowed to be set to any value, you can use ANY.", which to me was a hint that it was allowed to use in any context.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/15415 was merged, so this will be in next 2.1
